templates/server-ingress.yaml

```diff
@@ -0,0 +1,59 @@
+# Ingress for Vault cluster
+{{- if and (eq (.Values.server.ingress.enabled | toString) "true" ) (eq (.Values.global.enabled | toString) "true") }}
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: {{ template "vault.fullname" . }}
+  labels:
+    helm.sh/chart: {{ include "vault.chart" . }}
+    app.kubernetes.io/name: {{ include "vault.name" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+  annotations:
+    {{- if .Values.server.ingress.useNginx.enabled }}
```

Comment:
I appreciate making it easier to setup a specific ingress, but might we just accept annotations and document what annotations you might use for say Nginx? This would let the solution be more flexible.